### PR TITLE
Fix issue where initializeAll does not properly use initializer

### PR DESCRIPTION
**Pull request by schveiguy**
`TypeInfo_StaticArray.initializer` only returns an initializer for one element. In the case where `initializeAll` is given a range of static arrays of types with elaborate assign, random data would be used. This is also a memory corruption issue, since memcpy was reading past the end of the initializer. I tested the unit test before making the fix, and it was failing.

### CODE REVIEW

File: std/algorithm/mutation.d [hunk 1]
```diff
@@ -859,6 +878,27 @@ unittest
...
+        int x = 3;
+    }
+    Int[2] xs = [Int(1), Int(2)];
+    struct R {
+        bool done;
+        bool empty() { return done; }
+        ref Int[2] front() { return xs; }
+        void popFront() { done = true; }
+    }
+    /*writeln(xs);*/
```
Comment (DmitryOlshansky) [inline std/algorithm/mutation.d]: Debug print?

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by schveiguy** [comment]
@DmitryOlshansky yeah, copied from the bug report. I've removed it now.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**